File: rb/features.rb
# AiPresentationGenerator SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module AiPresentationGeneratorFeatures
  def self.make_feature(name)
    case name
    when "base"
      AiPresentationGeneratorBaseFeature.new
    when "ratelimit"
      AiPresentationGeneratorRatelimitFeature.new
    when "retry"
      AiPresentationGeneratorRetryFeature.new
    when "test"
      AiPresentationGeneratorTestFeature.new
    when "timeout"
      AiPresentationGeneratorTimeoutFeature.new
    else
      AiPresentationGeneratorBaseFeature.new
    end
  end
end

File: py/aipresentationgenerator_sdk/features.py
# AiPresentationGenerator SDK feature factory

from aipresentationgenerator_sdk.feature.base_feature import AiPresentationGeneratorBaseFeature
from aipresentationgenerator_sdk.feature.test_feature import AiPresentationGeneratorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AiPresentationGeneratorBaseFeature(),
        "test": lambda: AiPresentationGeneratorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
